src/path.c

```diff
@@ -0,0 +1,90 @@
+#include "path.h"
+
+static bool straight = true;
+static char *source;
+static enum movement *destination;
+
+/**
+ * @brief Translate the source path when coming from a straight movement.
+ *
+ * @return Number of movements translated from source.
```

Comment:
I would add "pointer iteration" somehow on comments, and maybe include it on the static function